packages/open-truss/src/configuration/engine-v1/Frame.tsx

```diff
@@ -126,20 +141,96 @@ function processProps({
     }
   }
 
+  eachComponentSignal(componentName, COMPONENTS, (propName, signalsType) => {
+    const viewPropValue =
+      viewProps === undefined ? undefined : viewProps[propName]
+    if (viewPropValue !== undefined) {
+      if (isSymbol(viewPropValue)) {
+        // If viewPropValue is a :symbol, use the signal by that name.
+        const signalName = parseSignalName(viewPropValue) ?? ''
+        const signal = signals[signalName]
+        if (signal) addSignalToProps(propName, signal, signalsType, newProps)
+      } else {
+        // Else the view prop is a scalar and we coerce the scalar into a signal type.
+        const signal = signalsType.parse(viewPropValue)
+        addSignalToProps(propName, signal, signalsType, newProps)
+      }
+    } else {
+      // Default lookup signal baesd on the name of the component prop.
+      const signal = signals[propName]
+      if (signal) {
+        addSignalToProps(propName, signal, signalsType, newProps)
+      }
+    }
+  })
+
   return {
-    data,
-    config,
     ...viewProps,
     ...newProps,
   }
 }
 
+function addSignalToProps(
+  propName: string,
+  signal: Signal<any>,
+  signalsType: SignalsZodType,
+  newProps: ComponentPropsReturnShape,
+): void {
+  const result = signalsType.safeParse(signal)
+  if (result.success) {
+    newProps[propName] = signal
+  } else {
+    throw new Error(`Signal passed to component is incorrect type.
+      Expected ${propName} to be type ${signalsType.description},
+      but got ${typeof signal}.
+    `)
+  }
+}
+
+function isSymbol(possibleSignal: any): boolean {
+  if (typeof possibleSignal !== 'string') return false
+  if (possibleSignal[0] !== ':') return false
```

Comment:
Could we use `String#startsWith`?

```suggestion
  if (!possibleSignal.startsWith(':')) return false
```